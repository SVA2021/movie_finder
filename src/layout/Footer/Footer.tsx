import {Box, Grid, Typography} from '@mui/material';
import {FC} from 'react';
import {StyledLink} from '../../components';

const LINKS = [
  {
    title: 'Backend API',
    href: 'https://kinopoiskapiunofficial.tech/',
  },
  {
    title: 'Портфолио SVA',
    href: 'https://portfolio-sva2021.vercel.app/',
  },
  {
    title: 'Github SVA',
    href: 'https://github.com/SVA2021',
  },
  {
    title: 'Code',
    href: 'https://github.com/SVA2021/movie_finder',
  },
]

export const Footer: FC = () => {
  return (
    <Box mt={2} sx={{flexGrow: 1}}>
      <Grid container>
        {
          LINKS.map((link, index) =>
            <Grid item xs={12} md={3} key={index}>
              <StyledLink href={link.href} title={link.title} />
            </Grid>
          )
        }
      </Grid>
      <Typography gutterBottom component={'p'} textAlign={'center'} color={'secondary'} >
        SVA	2023
      </Typography>
    </Box>
  )
};