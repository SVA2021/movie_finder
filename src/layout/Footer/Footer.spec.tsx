import {render, screen} from "@testing-library/react";
import {Footer} from "./Footer";

describe('render Footer', () => {

  const setUp = () => render(<Footer />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('footer')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/SVA 2023/i)).toBeInTheDocument();
  });

  it('check links', () => {
    setUp();
    expect(screen.queryAllByRole('link')).toHaveLength(4);
  });
})