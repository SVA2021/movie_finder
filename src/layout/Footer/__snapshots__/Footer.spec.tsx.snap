// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Footer snapshot test 1`] = `
<div
  class="MuiBox-root css-1xpmf4r"
  data-testid="footer"
>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
        data-testid="styled-link"
        href="https://kinopoiskapiunofficial.tech/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
        >
          Backend API
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
        data-testid="styled-link"
        href="https://portfolio-sva2021.vercel.app/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
        >
          Портфолио SVA
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
        data-testid="styled-link"
        href="https://github.com/SVA2021"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
        >
          Github SVA
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
        data-testid="styled-link"
        href="https://github.com/SVA2021/movie_finder"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
        >
          Code
        </p>
      </a>
    </div>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1o4n9zm-MuiTypography-root"
  >
    SVA 2023
  </p>
</div>
`;
